concepts/05 UI Components/DataGrid/20 Editing/43 Customize Edit Form/10 Appearance and Editing/10 Form Layout Customization.md

```diff
@@ -0,0 +1,189 @@
+        DxColumn,
+        DxFormItem,
+        DxLabel
+    } from 'devextreme-vue/data-grid';
+
+    export default {
+        components: {
+            DxDataGrid,
+            DxEditing,
+            DxColumn,
+            DxFormItem,
+            DxLabel
+        },
+        data() {
+            return {
+                columnEditorOptions: { height: 100 }
+            }
+        }
+        // ...
+    }
+    </script>
+
+##### React
+
+    <!-- tab: App.js -->
+    import React from 'react';
+    import 'devextreme/dist/css/dx.light.css';
+
+    import DataGrid, {
+        Editing,
+        Column,
+        FormItem,
+        Label
+    } from 'devextreme-react/data-grid';
+
+    const defaultEditorOptions = { height: 100 }
+
+    export default function App() {
+        return (
+            <DataGrid ... >
+                <Editing
+                    allowUpdating={true}
+                    mode="form"
+                />
+                <Column dataField="ID">
+                    <FormItem visible={false} />
+                </Column>
+                <Column dataField="Notes">
+                    <FormItem 
+                        colSpan={2}
+                        cssClass = "custom-class"
+                        editorType = "dxTextArea"
+                        editorOptions = "{ defaultEditorOptions }"
+                    >
+                        <Label location="top" />
+                    </FormItem>
+                </Column>
+            </DataGrid>
+        );
+    }
+
+---
+
+If you need to implement major changes (for example, arrange items and groups), use the **editing**.[form](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/editing/#form) property. Declare the required items for this object and specify the [dataField](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField) property. 
+
+DataGrid automatically generates editors with the related event handlers for the form items based on column settings and other DataGrid options like [onEditorPrepared](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#dataField). Do not use the [editCellTemplate](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editCellTemplate) or [editorOptions](/Documentation/ApiReference/UI_Components/dxDataGrid/Configuration/columns/#editorOptions) properties for items that are bound to **dataField** in the **editing.form** configuration. You can add unbound items just like for a regular [Form](/Documentation/ApiReference/UI_Components/dxForm/) component.
+
+The [Organize Simple Items](/Documentation/Guide/UI_Components/Form/Organize_Simple_Items/In_Groups/) topic illustrates how to organize form items in the Form component. You can use the same approach for the DataGrid editing form. Note that you should not specify the **formData** property because DataGrid uses its own mechanism under the hood to manage edited data.
```

Comment:
I thought, it is bad practice to write about incorrect behavior of our components. In our documentation we just recommend do not use some properties to avoid wrong behavior

Reply:
I understand. I will not insist -- I'm just thinking out loud that you should warn of possible consequences since you mention this fact in great enough detail.